# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  722d53af2a4ae4a92227fb80a8bf45402016a8c4
Backout bug 1959340 for causing bug 1975147: IMAP trash not emptied on exit.

diff --git a/mailnews/base/src/nsMsgAccountManager.cpp b/mailnews/base/src/nsMsgAccountManager.cpp
--- a/mailnews/base/src/nsMsgAccountManager.cpp
+++ b/mailnews/base/src/nsMsgAccountManager.cpp
@@ -141,12 +141,12 @@ static nsCOMPtr<nsIAsyncShutdownService>
   MOZ_RELEASE_ASSERT(service);
   return service;
 }
 
-static nsCOMPtr<nsIAsyncShutdownClient> GetAppShutdownConfirmed() {
+static nsCOMPtr<nsIAsyncShutdownClient> GetQuitApplicationGranted() {
   nsCOMPtr<nsIAsyncShutdownClient> barrier;
   nsresult rv =
-      GetShutdownService()->GetAppShutdownConfirmed(getter_AddRefs(barrier));
+      GetShutdownService()->GetQuitApplicationGranted(getter_AddRefs(barrier));
   MOZ_RELEASE_ASSERT(NS_SUCCEEDED(rv));
   MOZ_RELEASE_ASSERT(barrier);
   return barrier;
 }
@@ -195,9 +195,9 @@ nsresult nsMsgAccountManager::Init() {
     observerService->AddObserver(this, ABOUT_TO_GO_OFFLINE_TOPIC, true);
     observerService->AddObserver(this, "sleep_notification", true);
   }
 
-  GetAppShutdownConfirmed()->AddBlocker(
+  GetQuitApplicationGranted()->AddBlocker(
       this, NS_LITERAL_STRING_FROM_CSTRING(__FILE__), __LINE__,
       u"nsMsgAccountManager cleanup on exit"_ns);
   GetProfileBeforeChange()->AddBlocker(
       this, NS_LITERAL_STRING_FROM_CSTRING(__FILE__), __LINE__,
@@ -1747,9 +1747,9 @@ nsresult nsMsgAccountManager::CleanupOnE
       }
     }
   }
 
-  GetAppShutdownConfirmed()->RemoveBlocker(this);
+  GetQuitApplicationGranted()->RemoveBlocker(this);
 
   // Try to do this early on in the shutdown process before
   // necko shuts itself down.
   CloseCachedConnections();
@@ -1769,9 +1769,9 @@ nsMsgAccountManager::GetState(nsIPropert
 NS_IMETHODIMP
 nsMsgAccountManager::BlockShutdown(nsIAsyncShutdownClient* aClient) {
   nsAutoString name;
   aClient->GetName(name);
-  if (name.Equals(u"quit-application"_ns)) {
+  if (name.Equals(u"quit-application-granted"_ns)) {
     return CleanupOnExit();
   } else {
     // profile-before-change
     return Shutdown();
